src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/util/ManagedIndexUtils.kt

```diff
@@ -139,6 +159,17 @@ fun getDeleteManagedIndexRequests(
     }.map { deleteManagedIndexRequest(it.value.uuid) }
 }
 
+// if managed index exist but the index is not existing any more
+// then we should delete this managed index
+fun getManagedIndicesToDelete(
+    currentIndices: List<IndexMetadata>,
+    currentManagedIndexConfigs: Map<String, SweptManagedIndexConfig>
+): List<Index> {
+    return currentManagedIndexConfigs.filter { currentManagedIndex ->
+        !currentIndices.map { it.index.uuid }.contains(currentManagedIndex.key)
+    }.map { Index(it.value.index, it.value.uuid) }
+}
+
```

Comment:
Can change to something like below. Otherwise this looks very expensive once a cluster has thousands of indices to be converting that list to a map and checking if it contains the managed index on each iteration of a managed index.
```kotlin
// if managed index exist but the index is not existing any more
// then we should delete this managed index
fun getManagedIndicesToDelete2(
    currentIndices: List<IndexMetadata>,
    currentManagedIndexConfigs: Map<String, SweptManagedIndexConfig>
): List<Index> {
    val currentIndicesSet = currentIndices.map { it.index }.toSet()
    val managedIndicesSet = currentManagedIndexConfigs.values.map { Index(it.index, it.uuid) }.toSet() 
    return (managedIndicesSet - currentIndicesSet).toList()
}
```

Reply:
looks cool, I choose this!